core/src/main/java/flatgraph/GNode.java

```diff
@@ -51,4 +54,42 @@ public String toString() {
     public int hashCode() {
         return Objects.hash(graph, nodeKind, _seqId);
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I think this is very questionable. At least we'd need to check identity of the graphs for general equality checking; but we guarantee that only one node object is instantiated per logical node.

In other words, the default java object identity should do it.

Reply:
yeah I had some doubts too, I guess it's best to drop this and handle it in the DiffTool separately

Reply:
done